### ISSUE TITLE
API: Vote Here - GraphQL or JSON API

### ISSUE THREAD

**Issue by BenMorganIO** [ryal/ryal#23]
Please use a Github Reaction for your vote:

:tada: - [GraphQL](http://graphql.org/)
:heart: - [JSON API](http://jsonapi.org/)

---

We could use both since GraphQL prefers to work with `application/json` and `application/graphql`, while JSON API prefers the `application/vnd.api+json`. Since there's no overlap in content types, we should be able to serve both markets. This would be a huge :+1: for both EmberJS developers and Elm or React developers.

**Comment by hickscorp** [comment]
The mime-type shouldn't be part of the conversation: a graphql API could live on a different endpoint.

**Comment by yordis** [comment]
Could you add `Non-JSON API` or something similar, after using JSON API for a while I wouldn't use it again.

**Comment by James-Byrne** [comment]
I agree with @hickscorp . If we can be agnostic of this all the better. Otherwise I'll cast a vote for JSON API, but then I work with ember a lot :smile:

**Comment by yordis** [comment]
@James-Byrne that's exactly my problem with JSON API, it force you to an ecosystem and tooling everywhere.

I agree in some of the spec details but after using it without any Ember or X tool. It's really hard to work with and when you go out of the classic crud, then you start seeing special cases on JSON API.

Because of that, I would prefer GraphQL, at least the special cases are not a problem :kappa:

**Comment by asummers** [comment]
Random passerby: for JSONAPI, [ja_serializer](https://github.com/vt-elixir/ja_serializer) has worked very well for us, and having JSONAPI would allow this to be more easily integrated with our API. That said, it should just be "the web front end" to the larger application, so the project shouldn't have to care, just that it can express ingress and egress appropriately. Imagine someone wanting to put their OWN web front end in front of this logic, as an example with extra security.
